olp-cpp-sdk-dataservice-write/tests/StreamLayerClientImplTest.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gmock/gmock.h>
+#include <matchers/NetworkUrlMatchers.h>
+#include <mocks/CacheMock.h>
+#include <mocks/NetworkMock.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+#include <olp/dataservice/write/generated/model/ResponseOkSingle.h>
+#include "StreamLayerClientImpl.h"
+
+namespace {
+
+using namespace testing;
+using namespace olp;
+using namespace olp::dataservice::write;
+using namespace olp::tests::common;
+
+constexpr int64_t kTwentyMib = 20971520;
+const olp::client::HRN kHRN{"hrn:here:data:::catalog"};
+constexpr auto kLayerName = "layer";
+constexpr auto kContentType = "application/json";
+
+struct ResponseOkWithTraceIdFrom {
+  template <typename Request>
+  model::ResponseOkSingle operator()(Request request) {
+    model::ResponseOkSingle response;
+    response.SetTraceID(*request.GetTraceId());
+    return response;
+  }
+};
+
+MATCHER_P(WithTraceId, trace_id, "") {
+  return arg.GetTraceId() && *arg.GetTraceId() == trace_id;
+}
+
+class MockStreamLayerClientImpl : public StreamLayerClientImpl {
+ public:
+  using StreamLayerClientImpl::StreamLayerClientImpl;
+
+  MOCK_METHOD(PublishSdiiResponse, IngestSdii,
+              (model::PublishSdiiRequest request,
+               client::CancellationContext context),
+              (override));
+};
+
+class StreamLayerClientImplTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    cache_ = std::make_shared<CacheMock>();
+    network_ = std::make_shared<NetworkMock>();
+
+    settings_.network_request_handler = network_;
+    settings_.cache = cache_;
+    settings_.task_scheduler =
+        olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler(1);
+  }
+
+  void TearDown() override {
+    settings_.network_request_handler.reset();
+    settings_.cache.reset();
+
+    network_.reset();
+    cache_.reset();
+  }
+
+  std::shared_ptr<CacheMock> cache_;
+  std::shared_ptr<NetworkMock> network_;
+  olp::client::OlpClientSettings settings_;
+};
+
+TEST_F(StreamLayerClientImplTest, PublishSdii) {
```

Comment:
```suggestion
TEST_F(StreamLayerClientImplTest, GivenGoodRequestWhenPublishSdiiThenOKIsReceived) {
```
A few comments on that test:

1. Would splitting this test make sense? It's better to test one thing per test case, isn't it?
2. When split, I'd use the "[GivenWhenThen](https://martinfowler.com/bliki/GivenWhenThen.html)" naming technique used in the suggestion. But that's optional as it requires a buy-in from the whole team.

Reply:
Please nevermind this comment. Andrei explained to me that we have a different approach described at https://confluence.in.here.com/pages/viewpage.action?pageId=789959628